### PR TITLE
Fix issue 6433

### LINKED ISSUE #6433: Non-termination in CFG construction
Related to #6430 

On the following test input, CFG construction does not terminate for the `main` method:
```java
import java.io.*;
import java.net.URISyntaxException;
import java.security.*;
import java.security.cert.CertificateException;
import javax.net.ssl.*;

class Test {
  public static void SSLFollowedByRegularConnection()
      throws IOException,
          InterruptedException,
          NoSuchAlgorithmException,
          KeyManagementException,
          KeyStoreException,
          UnrecoverableKeyException,
          CertificateException,
          URISyntaxException {}

  public static void main(String[] args)
      throws IOException,
          InterruptedException,
          NoSuchAlgorithmException,
          KeyManagementException,
          KeyStoreException,
          UnrecoverableKeyException,
          CertificateException,
          URISyntaxException {
    SSLFollowedByRegularConnection();
    while (true) {}
  }
}
```

(Probably could be minimized further.)  Execution gets stuck in the `org.checkerframework.dataflow.cfg.builder.CFGTranslationPhaseThree#mergeConsecutiveBlocks` routine.

### PR DESCRIPTION
**Pull request by msridhar**
Fixes #6433 

Fix taken from https://github.com/eisop/checker-framework/commit/faf33e68377774fa276fad8e8717b8bfee48c42a and adapted to fix merge conflicts

### CODE REVIEW

File: dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseThree.java [hunk 1]
```diff
@@ -164,6 +167,10 @@ protected static void mergeConsecutiveBlocks(ControlFlowGraph cfg) {
           Block succ = b.getRegularSuccessor();
           if (succ.getType() == BlockType.REGULAR_BLOCK) {
             RegularBlockImpl rs = (RegularBlockImpl) succ;
+            if (rs.getRegularSuccessor() == rs) {
+              // An infinite loop, do not try to merge.
```
Comment (mernst) [inline dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseThree.java:171]: I think this is an infinite loop at run time, but just a loop in the CFG.  Is that correct?

**Review by mernst — APPROVED** [review]
Thanks, Manu and Werner.

File: dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseThree.java (hunk 1 above)
Comment (msridhar) [reply dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseThree.java:171]: I _think_ a `RegularBlock` can only have one successor in the CFG.  So this is both a loop in the CFG and an infinite loop at run time.  Should I update the comment to that effect?

Comment (mernst) [reply dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseThree.java:171]: You are right about `RegularBlock` successors.  A small tweak to the comment would be good -- "infinite loop" doesn't really apply to a CFG (or other) graph, so the wording took me aback.

Thanks again.

**Comment by msridhar** [comment]
@mernst I fixed the comment, so this needs another stamp from you

**Review by mernst — APPROVED** [review]

**Pull request merged**